server/sockets/list.js

```diff
@@ -413,42 +433,62 @@ const changeListType = (socket, dashboardClients, cohortClients, listClients) =>
 
           if (dashboardClients.size > 0) {
             if (type === ListType.LIMITED && removedViewers) {
-              emitForMany(
-                removedViewers,
-                dashboardClients,
-                socket,
-                ListActionTypes.DELETE_SUCCESS,
-                listId
-              );
+              removedViewers.forEach(id => {
+                const userId = id.toString();
+
+                if (dashboardClients.has(userId)) {
+                  socket.broadcast
+                    .to(dashboardClients.get(userId))
+                    .emit(ListActionTypes.DELETE_SUCCESS, listId);
+                }
+              });
```

Comment:
please check if dashboardClients.size > 0 is necessary, cause if you do like this:

```
if(dashboardClient.has(userid) && type === ListType.Limited && removedViewers) { 
   ...socket()

```
you can probably get rid of nested if and checking size

Reply:
I removed checking the size and refactored the code but a little bit different.